website/docs/tutorials/deploying-java-app-with-database.md

```diff
@@ -0,0 +1,250 @@
+
+In this example we will use odo to manage a sample [Java MicroServices JPA application](https://github.com/OpenLiberty/application-stack-samples.git).
+
+1. Clone the sample application to your system:
+    ```shell
+    git clone https://github.com/OpenLiberty/application-stack-samples.git
+    ```
+
+2. Go to the sample JPA app directory:
+    ```shell
+    cd ./application-stack-samples/jpa
+    ```
+
+3. Initialize the application:
+    ```shell
+    odo create java-openliberty mysboproj
+    ```
+   `java-openliberty` is the type of your application and `mysboproject` is the name of your application.
+
+4. Deploy the application to the cluster:
+    ```shell
+    odo push --show-log
+    ```
+
+5. The application is now deployed to the cluster - you can view the status of the cluster, and the application test results by streaming the cluster logs of the component that we pushed to the cluster in the previous step.
+    ```shell
+    odo log --follow
+    ```
+
+    Notice the failing tests due to an `UnknownDatabaseHostException`:
+    ```shell
+    [INFO] [err] java.net.UnknownHostException: ${DATABASE_CLUSTERIP}
+    [INFO] [err]    at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:220)
+    [INFO] [err]    at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
+    [INFO] [err]    at java.base/java.net.Socket.connect(Socket.java:609)
+    [INFO] [err]    at org.postgresql.core.PGStream.<init>(PGStream.java:68)
+    [INFO] [err]    at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:144)
+    [INFO] [err]    ... 86 more
+    [ERROR] Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 0.706 s <<< FAILURE! - in org.example.app.it.DatabaseIT
+    [ERROR] testGetAllPeople  Time elapsed: 0.33 s  <<< FAILURE!
+    org.opentest4j.AssertionFailedError: Expected at least 2 people to be registered, but there were only: [] ==> expected: <true> but was: <false>
+    at org.example.app.it.DatabaseIT.testGetAllPeople(DatabaseIT.java:57)
+    
+    [ERROR] testGetPerson  Time elapsed: 0.047 s  <<< ERROR!
+    java.lang.NullPointerException
+    at org.example.app.it.DatabaseIT.testGetPerson(DatabaseIT.java:41)
+    
+    [INFO]
+    [INFO] Results:
+    [INFO]
+    [ERROR] Failures:
+    [ERROR]   DatabaseIT.testGetAllPeople:57 Expected at least 2 people to be registered, but there were only: [] ==> expected: <true> but was: <false>
+    [ERROR] Errors:
+    [ERROR]   DatabaseIT.testGetPerson:41 NullPointer
+    [INFO]
+    [ERROR] Tests run: 2, Failures: 1, Errors: 1, Skipped: 0
+    [INFO]
+    [ERROR] Integration tests failed: There are test failures.
+    ```
+   Note: This error will be fixed at a later stage in the tutorial when we connect a database instance to this application.
+
+6. You can also create a URL with `odo` to access the application:
+    ```shell
+    odo url create --host $(minikube ip).nip.io
```

Comment:
@dharmit We will need to enable ingress and ingress-dns addons on minikube for this to work, won't we?

Reply:
Yes. But that should be covered in the Kubernetes cluster setup guide in Getting Started section.